src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -200,6 +210,12 @@ mkLevelsCache reg lvls = do
 --
 -- * Keep the cache feature, but force a rebuild every once in a while, e.g.,
 --   once in every 100 lookups.
+--
+-- TODO: rebuilding the cache can invalidate blob references if the cache was
+-- holding the last reference to a run. This is not really a problem of just the
+-- caching approach, but allowing merges to finish early. We should come up with
+-- a solution to keep blob references valid until the next /update/ comes along.
+-- Lookups should no invalidate blob erferences.
```

Comment:
Rebuilding the cache on `lookups` is another thing that is something we should implement, but I also estimate it brings quite a bit of complexity. As such, we should fix this TODO it in a follow-up PR -- the current version shows that the naive approach is working

Reply:
How does merges finishing early come into play? Ah, you mean that they finish because of merging done from another table handle, and that change being reflected immediately in this table handle?

Reply:
> Ah, you mean that they finish because of merging done from another table handle, and that change being reflected immediately in this table handle?

Indeed. If updates to a different table cause the merge to be completed, then the input runs will be closed, and all of the blob references weakly referencing those input runs will be invalidated